tools/autograd/load_derivatives.py

```diff
@@ -207,8 +217,14 @@ def unzip(xs):
                                .format(i, defn_name, x, y))
 
     derivatives, args_with_derivatives, non_differentiable_arg_names = set_up_derivatives(defn_name, defn, canonical)
-    return create_autograd_function(defn_name, derivatives, args_with_derivatives, non_differentiable_arg_names,
-                                    signature, canonical, output_differentiability)
+    autograd_fn = None
+
+    # only create an autograd function if we are actually going to calculate a derivative
```

Comment:
Does this mean 'only create an autograd function, if we actually plan to use it'?

Reply:
Or "don't create an autograd function, if there is no derivative available", which means it can't possibly be differentiable.

Reply:
It means the former, although I prefer the wording in the diff.

The later i slightly wrong because if no derivatives are specified it can still fall through, i.e. be differentiable.

Reply:
Well, we can calculate a derivative and have an implementation, but not provide it as part of autograd or plan to do so.

The term "derivative" here is confusing, because there's operators with explicit derivative implementations (i.e. xyz_backward) and some operators where the derivative is a result of the fact that they're implemented as a string of already differentiable operators. 

It'd be nice to have a clear distinction between those cases.